spark/src/test/scala/org/apache/spark/sql/delta/GeneratedColumnSuite.scala

```diff
@@ -1724,6 +1724,30 @@ trait GeneratedColumnSuiteBase extends GeneratedColumnTest {
     }
   }
 
+  test("MERGE INSERT with schema evolution on different name case") {
```

Comment:
What's the relevance of generated columns for this issue? Should we also add test for identity columns?

Reply:
Not particularly relevant except that the code path for generated columns happened to have an assertion that was triggered by this issue. We land back on our feet in cases where that assertion is not triggered (at least I didn't find a problematic case). Since that's the test that triggered the issue, it's relevant to include it here